docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,273 @@
+
+* Don't split senctences across lines but keep them on one single line.
+* Start a new line for each and every sentence.
+* Follow the https://docs.microsoft.com/en-us/style-guide/welcome/[Microsoft Writing Style Guide].
+
+[NOTE]
+====
+The Microsoft Writing Style is being enforced with https://github.com/errata-ai/vale[Vale].
+The implementation used can be found at https://github.com/errata-ai/Microsoft[errata-ai/Microsoft].
+====
+
+== Jsonnet Style
+
+== Syntactic Style
+
+=== Auto Formatting
+
+Use `jsonnet fmt` to format files. This will fix basic style errors.
+
+=== Variable Declaration
+
+* Variables should be named in `camelCase` style, and should have self-evident names.
++
+[source,jsonnet]
+----
+local serverPort = 1000;
+local clientPort = 2000;
+----
+
+* Prefer `local` to `::` syntax for private/local variables.
+  Unlike `::`, variables defined with `local` can't be overridden by children, nor accessed by other files.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  local myVariable = 3,
+  result: myVariable + 1,
+
+  // INCORRECT
+  myVariable:: 3,
+  result: $.myVariable + 1,
+}
+----
+
+=== Line Length
+
+* Limit lines to 120 characters.
+* The only exceptions are import statements and URLs (although even for those, try to keep them under 120 chars).
+
+=== Spacing and Indentation
+
+* Put one space before and after operators.
++
+[source,jsonnet]
+----
+local c = a + b;
+----
+
+* Put one space after commas.
++
+[source,jsonnet]
+----
+["a", "b", "c"] // CORRECT
+
+["a","b","c"] // INCORRECT
+----
+
+* Put one space after colons.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  foo:: "bar",
+  baz: "taz",
+  { hello: "world" },
+
+  // INCORRECT
+  foo :: "bar",
+  baz:"taz",
+  { hello : "world" },
+}
+----
+
+* Put one space or line brake after opening curly brackets and before closing curly brackets.
++
+[source,jsonnet]
+----
+// CORRECT
+local foo = { hello: "world" };
+local bar = {
+  hello: "world",
+};
+
+// INCORRECT
+local baz = {hello: "world"};
+----
+
+* Use 2-space indentation in general.
+
+* Only method or class parameter declarations use 4-space indentation, to visually differentiate parameters from method body.
++
+[source,jsonnet]
+----
+// CORRECT
+local multiply(
+    number1,
+    number2) = {
+  result: number1 * number 2
+}
+----
+
+* Omit vertical alignment.
+  They draw attention to the wrong parts of the code and result in hard to review pull requests due to the white space changes.
++
+[source,jsonnet]
+----
+// Don't align vertically
+local plus     = "+";
+local minus    = "-";
+local multiply = "*";
+
+// Do the following
+local plus = "+";
+local minus = "-";
+local multiply = "*";
+----
+
+=== Blank Lines (Vertical Whitespace)
+
+* A single blank line appears:
+** Within method bodies, as needed to create logical groupings of statements.
+** Optionally before the first member or after the last member of a class or method.
+* Use one or two blank line(s) to separate class definitions.
```

Comment:
More generally, use one or two blank line(s) to separate logical blocks in files. Those blocks can be single function definitions or groups of local variables that semantically belong together.